internal/controller/pod_controller_test.go

```diff
@@ -199,11 +209,121 @@ var _ = Describe("CatGate controller", func() {
 					numSchedulable += 1
 				}
 			}
+			// 2 pods are already running, so 6 (2 + 2*2) pods should be scheduled
 			g.Expect(numSchedulable).To(Equal(6))
 		}).Should(Succeed())
 	})
 
-	// TODO: add a test where one node has multiple running pods and only one additional pod would be scheduled.
+	It("should allow scheduling of additional pods when multiple pods are running on a single node", func() {
+		testName := "multiple-pods-running-on-single-node"
+		namespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: testName,
+			},
+		}
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).NotTo(HaveOccurred())
+
+		for i := 0; i < 3; i++ {
+			createNewPod(i, testName)
+		}
+		for i := 0; i < 1; i++ {
+			createNewNode(i, testName)
+		}
+
+		nodes := &corev1.NodeList{}
+		err = k8sClient.List(ctx, nodes, &client.ListOptions{Namespace: testName})
+		Expect(err).NotTo(HaveOccurred())
+
+		pods := &corev1.PodList{}
+		Eventually(func(g Gomega) {
+			err := k8sClient.List(ctx, pods, &client.ListOptions{Namespace: testName})
+			g.Expect(err).NotTo(HaveOccurred())
+			numSchedulable := 0
+			for _, pod := range pods.Items {
+				if !existsSchedulingGate(&pod) {
+					numSchedulable += 1
+				}
+			}
+			// no pod is already running, so 1 pods should be scheduled
+			g.Expect(numSchedulable).To(Equal(1))
+		}).Should(Succeed())
+		scheduleSpecificNodeAndStartOnePod(testName, nodes.Items[0].Name)
+
+		Eventually(func(g Gomega) {
+			err := k8sClient.List(ctx, pods, &client.ListOptions{Namespace: testName})
+			g.Expect(err).NotTo(HaveOccurred())
+			numSchedulable := 0
+			for _, pod := range pods.Items {
+				if !existsSchedulingGate(&pod) {
+					numSchedulable += 1
+				}
+			}
+			// 1 pod is already running, so 3(1 + 1*2) pods should be scheduled
+			g.Expect(numSchedulable).To(Equal(3))
+		}).Should(Succeed())
+		scheduleSpecificNodeAndStartOnePod(testName, nodes.Items[0].Name)
+
+		Eventually(func(g Gomega) {
+			err := k8sClient.List(ctx, pods, &client.ListOptions{Namespace: testName})
+			g.Expect(err).NotTo(HaveOccurred())
+			numSchedulable := 0
+			for _, pod := range pods.Items {
+				if !existsSchedulingGate(&pod) {
+					numSchedulable += 1
+				}
+			}
+			// 2 pods are already running, so 3 pods should be scheduled
```

Comment:
The situation includes a special condition, so it would be beneficial to write that explicitly.

```suggestion
			// 2 pods are already running on a same node, so 3 pods should be scheduled
```